packages/react-query/src/useBaseQuery.ts

```diff
@@ -82,26 +82,27 @@ export function useBaseQuery<
       ),
   )
 
+  const [_, setForceUpdate] = React.useState(0)
+
   const result = observer.getOptimisticResult(defaultedOptions)
```

Comment:
I wanted to use a `useState()` here but it made the blast radius really big and broke `useSuspense()`'s test etc

Reply:
If we get more errors, I'd bet this could be a culprit as it's reading results synchronously from the observer

We should maybe have a `const [result, setResult]` rather than a force update, but I'd kinda want to do this as a follow-up if we do it as it changes all `useSuspense()`-test because it'll take an extra render before the component is suspended